lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -182,6 +210,35 @@ def ensure_present(self):
         else:
             self.create()
 
+    def get_arn(self):
+        region, ec2_url, aws_connect_params = get_aws_connection_info(self.module, boto3=True)
+        if region:
+            client = boto3_conn(self.module, conn_type='client', resource='sts',
+                                region=region, endpoint=ec2_url, **aws_connect_params)
+        else:
+            self.module.fail_json(msg="region must be specified")
+        instance_counts = {}
+        response = client.get_caller_identity()
+        user_id = response['Account']
+        arn = "arn:aws:elasticache:" + region + ":" + user_id + ":cluster:" + self.name
+        return arn
+
+    def compare_and_update_tags(self):
+        if self.tags:
```

Comment:
This check shouldn't be required if `self.tags` defaults to the empty dict - the else clause would be taken care of in the same way using compare_aws_tags.